src/contracts/arkcontract.ts

```diff
@@ -0,0 +1,234 @@
+import { hex } from "@scure/base";
+import { Contract } from "./types";
+import { contractHandlers } from "./handlers";
+
+/**
+ * Prefix for arkcontract strings.
+ */
+const ARKCONTRACT_PREFIX = "arkcontract";
+
+/**
+ * Encode a contract to the arkcontract string format.
+ *
+ * Format: arkcontract={type}&{key1}={value1}&{key2}={value2}...
+ *
+ * This format is compatible with NArk and allows contracts to be
+ * shared/imported across different Ark implementations.
+ *
+ * @example
+ * ```typescript
+ * const contract: Contract = {
+ *   id: "abc123",
+ *   type: "vhtlc",
+ *   params: { sender: "ab12...", receiver: "cd34...", ... },
+ *   // ...
+ * };
+ *
+ * const encoded = encodeArkContract(contract);
+ * // "arkcontract=vhtlc&sender=ab12...&receiver=cd34...&..."
+ * ```
+ */
+export function encodeArkContract(contract: Contract): string {
+    const parts: string[] = [`${ARKCONTRACT_PREFIX}=${contract.type}`];
+
+    // Add all params
+    for (const [key, value] of Object.entries(contract.params)) {
+        parts.push(`${encodeURIComponent(key)}=${encodeURIComponent(value)}`);
+    }
+
+    // Add runtime data if present
+    if (contract.data) {
+        for (const [key, value] of Object.entries(contract.data)) {
+            parts.push(
+                `${encodeURIComponent(key)}=${encodeURIComponent(value)}`
+            );
+        }
+    }
+
+    return parts.join("&");
+}
+
+/**
+ * Parsed result from decoding an arkcontract string.
+ */
+export interface ParsedArkContract {
+    /** Contract type */
+    type: string;
+
+    /** All key-value pairs from the string */
+    data: Record<string, string>;
+}
```

Comment:
```suggestion
export interface ParsedArkContract<T> {
    /** Contract type */
    type: string;

    /** All key-value pairs from the string */
    data: T
}
```
We can use generics so we can type the data (less probability of dev errors)